src/When.php

```diff
@@ -460,48 +461,63 @@ public function getOccurrencesBetween($startDate, $endDate, $limit=NULL) {
             return $occurrences;
         }
 
-        self::prepareDateElements(false);
+        // if existing UNTIL < startDate - we have nothing
+        if (isset($this->until) && $this->until < $startDate) {
+            return $occurrences;
+        }
+        // prevent unnecessary leg-work - our endDate is our new UNTIL
+        elseif (!isset($this->until)) {
+            $this->until = $endDate;
+        }
 
-        list($startDate, $endDate) = $this->findDateRangeOverlap($startDate, $endDate);
+        $this->generateOccurrences();
+        $all_occurrences = $this->occurrences;
 
-        // If we have a defined $count, we need to test from $this->startDate to ensure we stop at $count.
-        if ($this->count and ($startDate > $this->startDate)) {
-            $max_occurrences = $this->count - $this->countOccurrencesBefore($startDate);
-            if ($max_occurrences <= 0) {
-                return $occurrences;
-            }
-        }
-        else {
-            $max_occurrences = $this->count;
+        // nothing found in $this->generateOccurrences();
+        if (empty($all_occurrences)) {
+            return $occurrences;
         }
 
-        if ($limit) {
-            if (! $max_occurrences || ($limit < $max_occurrences)) {
-                $max_occurrences = $limit;
+        $last_occurrence = end($all_occurrences);
+
+        // if our last occurrence is is before our startDate, we have nothing
+        if ($last_occurrence < $startDate) {
```

Comment:
Hmm that change is causing several other breaks in the phpunit tests. I will have to investigate further. Good catch though, I have written a test for this, and it currently fails as expected: 
https://gist.github.com/Blindmikey/d9d16caabed9cc18ec7d01b5abea5bf7